### PR TITLE
functional-tester: limit lease creation in lease stresser

### PR DESCRIPTION
**Pull request by fanminshi**
too many leases created during leases stressing phase is causing compaction to timeout.
bound leases creation solves compacting issue.

### CODE REVIEW

**Comment by fanminshi** [comment]
related the compaction issue in https://github.com/coreos/etcd/issues/6685

**Comment by heyitsanthony** [comment]
This should probably be done with `rate.Limiter`

File: tools/functional-tester/etcd-tester/lease_stresser.go [hunk 1]
```diff
@@ -161,14 +161,12 @@ func (ls *leaseStresser) Stress() error {
...
 	ls.restartKeepAlives()
-	for ls.ctx.Err() == nil {
-		plog.Debugf("creating lease on %v ", ls.endpoint)
-		ls.createLeases()
-		plog.Debugf("done creating lease on %v ", ls.endpoint)
-		plog.Debugf("dropping lease on %v ", ls.endpoint)
-		ls.randomlyDropLeases()
-		plog.Debugf("done dropping lease on %v ", ls.endpoint)
-	}
+	plog.Debugf("creating lease on %v ", ls.endpoint)
```
Comment (gyuho) [inline tools/functional-tester/etcd-tester/lease_stresser.go]: @fanminshi Let's strip extra whitespace at the end `%v "`

**Comment by xiang90** [comment]
@fanminshi Probably we can try limiter to limit this to a relatively low rate. like 1 round per 5 seconds or so.

File: tools/functional-tester/etcd-tester/lease_stresser.go (hunk 1 above)
Comment (fanminshi) [inline tools/functional-tester/etcd-tester/lease_stresser.go]: kk

**Comment by fanminshi** [comment]
All fixed. PTAL @xiang90 @gyuho

**Comment by xiang90** [comment]
let's make two pr. one for the limit thing; one for the lease update. thanks!

File: tools/functional-tester/etcd-tester/main.go [hunk 2]
```diff
@@ -53,6 +53,8 @@ func main() {
 	externalFailures := flag.String("external-failures", "", "specify a path of script for enabling/disabling an external fault injector")
 	flag.Parse()
 
+	capnslog.SetGlobalLogLevel(capnslog.DEBUG)
```
Comment (gyuho) [inline tools/functional-tester/etcd-tester/main.go]: Shouldn't this be configurable? /cc @xiang90

**Comment by gyuho** [comment]
Maybe another PR for log level configuration

File: tools/functional-tester/etcd-tester/main.go (hunk 2 above)
Comment (fanminshi) [inline tools/functional-tester/etcd-tester/main.go]: agreed

Comment (xiang90) [inline tools/functional-tester/etcd-tester/main.go]: i guess this is just for his debugging purpose. and it should be reverted.

**Comment by fanminshi** [comment]
all fixed. PTAL @xiang90 @gyuho @heyitsanthony

File: tools/functional-tester/etcd-tester/lease_stresser.go [hunk 3]
```diff
@@ -115,10 +118,13 @@ func newLeaseStresserBuilder(s string, lsConfig *leaseStressConfig) leaseStresse
 		}
 	case "default":
 		return func(mem *member) Stresser {
+			// limit lease stresser to run 1 round per second
+			l := rate.NewLimiter(1, 1)
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/lease_stresser.go]: define 1 and 1 as const somewhere?

**Comment by xiang90** [comment]
lgtm

File: tools/functional-tester/etcd-tester/lease_stresser.go [hunk 4]
```diff
@@ -170,13 +176,16 @@ func (ls *leaseStresser) Stress() error {
 func (ls *leaseStresser) run() {
 	defer ls.runWg.Done()
 	ls.restartKeepAlives()
-	for ls.ctx.Err() == nil {
-		plog.Debugf("creating lease on %v ", ls.endpoint)
+	for {
+		if err := ls.rateLimiter.Wait(ls.ctx); err == context.Canceled {
```
Comment (fanminshi) [inline tools/functional-tester/etcd-tester/lease_stresser.go]: That's right

**Pull request merged**